ROADMAP.md

```diff
@@ -1,53 +1,64 @@
-# Roadmap
+# Roadmap <!-- omit in toc -->
 
 The libp2p project is creating a set of modular, loosely-coupled building blocks
 (based on a shared core) for modern peer-to-peer networking. Our goal is to
 enable a new generation of decentralized applications in which network nodes are
-full peers, providing and consuming data, and routing traffic for one another,
-all without the need for centralized infrastructure points or reliance on
-third-party ownership of data.
-
-**Table of Contents**
-
-- [Roadmap](#roadmap)
-    - [Visionary](#visionary)
-        - [🖧 Decentralizing networks](#🖧-decentralizing-networks)
-        - [🤫 A spyproof libp2p](#🤫-a-spyproof-libp2p)
-        - [📱 libp2p in mobile devices](#📱-libp2p-in-mobile-devices)
-        - [💡 libp2p in IoT](#💡-libp2p-in-iot)
-        - [🎓 libp2p as a platform for Networks Research & Innovation](#🎓-libp2p-as-a-platform-for-networks-research--innovation)
-        - [🚑 Self-healing networks](#🚑-self-healing-networks)
-        - [📮 Offline message queue / postbox](#📮-offline-message-queue--postbox)
-        - [🤖 libp2p as a WASM library](#🤖-libp2p-as-a-wasm-library)
-    - [Evolve](#evolve)
-        - [✈️ WebTransport](#✈️-webtransport)
-        - [⏱ Full Observability](#⏱-full-observability)
-        - [🧪 Automated compatibility testing](#🧪-automated-compatibility-testing)
-        - [Stream Migration Protocol](#stream-migration-protocol)
-        - [WebRTC](#webrtc)
-        - [🤝 Low latency, efficient connection handshake](#🤝-low-latency-efficient-connection-handshake)
-        - [🛣️ Peer Routing Records](#🛣️-peer-routing-records)
-        - [🗣️ Polite peering](#🗣️-polite-peering)
-        - [🧱 Composable routing](#🧱-composable-routing)
-        - [💣 Attack resistance, threat models and security](#💣-attack-resistance-threat-models-and-security)
-        - [📈 Proving we are scalable and interoperable](#📈-proving-we-are-scalable-and-interoperable)
-        - [🌡️ Telemetry protocol](#🌡️-telemetry-protocol)
-        - [📩 Message-oriented transports](#📩-message-oriented-transports)
-        - [☎️ Reducing the dial fail rate](#️-reducing-the-dial-fail-rate)
-        - [🔀 Peer exchange protocol](#🔀-peer-exchange-protocol)
-        - [🏹 RPC and other common node communication patterns](#🏹-rpc-and-other-common-node-communication-patterns)
-    - [Done](#done)
-        - [🕸 Hole punching on TCP and QUIC](#🕸-hole-punching-on-tcp-and-quic)
+full peers, provide and consume data, and route traffic for one another,
+all without the need for centralized infrastructure or reliance on
+third-party data ownership.
```

Comment:
```suggestion
third-party data ownership.

This "roadmap" doesn't answer what functionality will land in which libp2p implementation by when.  For specific sequencing and dates, see implementation roadmaps:
* libp2p-wide test-plans:
* go: 
* rust: 
* js: 
* (other implementations: feel free to add)
```

Given this roadmap doesn't provide any sense of date/time, I'd like to link where someone can learn this.  You don't have to take my suggestion as is - it's just an idea.

I think we're ok to link to ROADMAP.md files even if they don't exist yet as I'm assuming they'll all get merged by end of the week.

Reply:
You might have missed it but it's under this section
```
### Implementation Roadmaps
- [go-libp2p](https://github.com/libp2p/go-libp2p/blob/master/ROADMAP.md)
- [js-libp2p](https://github.com/libp2p/js-libp2p/blob/master/ROADMAP.md)
- [rust-libp2p](https://github.com/libp2p/rust-libp2p/blob/master/ROADMAP.md)
```

I will add that supporting text

Reply:
Cool.  I think this commit came in after I had pulled up the PR.

Please also get the test-plan up at the top of the list given it's important.

Your call, but I don't think it hurts to add a note inviting other implementations to add their roadmaps.  I don't want to give the impression that libp2p is only go, js, and rust.